src/test/java/io/jenkins/plugins/opentelemetry/JenkinsOtelPluginNoConfigurationTest.java

```diff
@@ -0,0 +1,108 @@
+package io.jenkins.plugins.opentelemetry;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.nullValue;
+
+import hudson.model.Result;
+import io.jenkins.plugins.opentelemetry.init.StepExecutionInstrumentationInitializer;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporterProvider;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import org.hamcrest.Matchers;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.LogRecorder;
+
+public class JenkinsOtelPluginNoConfigurationTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public BuildWatcher buildWatcher = new BuildWatcher();
+
+    /**
+     * As the JVM and classes are loaded only once for the whole test, {@link SynchronousNonBlockingStepExecution#getExecutorService()} augments only once. The current boolean keeps track of the augmentation status.
+     */
+    private static boolean augmented = false;
```

Comment:
I am not sure what the goal of this boolean; it depends on the order of execution of the test, and it will be false/true when the test arrives at the call of `checkAugmentation`, I mean you have test A and B both call `checkAugmentation`, and `augmented` is false when the class is intanciated. If the test A run calls `checkAugmentation` with `augmented=false` and changes the value to true, the assert passes. Then, the test B will call `checkAugmentation` with `augmented=true`. If you invert the order of the test (B then A), B will call `checkAugmentation` with `augmented=false` and A will call `checkAugmentation` with `augmented=true` that's completely the opposite.

Reply:
Yes, it's the purpose of this boolean, we just want to check augmentation happened once.